blog "I used to think product people aren't needed" (#2783)
PostHog/posthog.com: contents/blog/product-360.md

---

> date: 2022-01-18

Reviewer: Does this need to be updated to today's date?

Reviewer 2: It does, yes. Thank you

> title: I used to think product people aren't needed

Reviewer 2: Just thinking about punching this title up a bit. A few options:

1. I used to think product people aren't needed. I was wrong
2. How I learned I was wrong about product people
3. Why coders make the best product people

I'm inclined to (1). Just adding the bit on the end just adds a bit more bite, I think.

Author: Agreed with 1

> title: I used to think product people aren't needed. I was wrong

Reviewer: Title suggestion - _I used to think you don't need product people. I was wrong._ 

(ie. make more active, and add full stop at the end)

Reviewer 2: agreed

hideAnchor: true
categories: ["CEO diaries", "Inside PostHog", "Engineering"]
author: ["james-hawkins"]
featuredImage: ../images/blog/product-people.png
featuredImageType: full
---

When PostHog was fresh out of YC last year, there were two things we focused on: (i) build and (ii) talk to customers. Hiring a product person felt like it would slow us down. Engineers on our team would speak to our users (also engineers) - surely they'd know what's best to prioritize? We didn't want our engineers to be "shielded" from our customers.

By [PostHog's series B](/blog/15-million-series-b), our team showed me that I was totally wrong. Here's why that was the case, what we did about it, and the results.

## Why I used to think this way

My previous experience of product management:

* Here's a list of features the enterprise sales team wants
* How much pipeline does each of these features influence?
* _Builds as much of the above as possible whilst trying to manage existing customer requests_

Lead to:

* Lots of initial enterprise contracts
* A product users find frustrating to use
* No word of mouth growth
* Customer churn in the long term

## It starts with expectation setting

> <a href="https://imgflip.com/i/61pr1n"><img src="https://i.imgflip.com/61pr1n.jpg" title="made at imgflip.com"/></a><div><a href="https://imgflip.com/memegenerator">from Imgflip Meme Generator</a></div><br />

Reviewer: I think this meme has been mis-used - the point of it is that Morpheus is sharing a mindblowing insight with you. Reading this, it sounds like you are saying 'The insight we found is that the board your product manager'.  

Something like this I think makes the point better:
<img width="474" alt="Screenshot 2022-02-03 at 12 20 04" src="https://user-images.githubusercontent.com/70321811/152341659-77acbc63-5b9c-4c7f-8233-971216e25c83.png">

Reviewer 2: @jamesefhawkins happy to use this one? I'll upload this as I'd prefer not to load stuff directly from imgflip

Reviewer 2: I've switching this in now

---
date: 2022-01-18
title: I used to think product people aren't needed. I was wrong
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ["CEO diaries", "Inside PostHog", "Engineering"]
author: ["james-hawkins"]
featuredImage: ../images/blog/product-people.png
featuredImageType: full
---

When PostHog was fresh out of YC last year, there were two things we focused on: (i) build and (ii) talk to customers. Hiring a product person felt like it would slow us down. Engineers on our team would speak to our users (also engineers) - surely they'd know what's best to prioritize? We didn't want our engineers to be "shielded" from our customers.

By [PostHog's series B](/blog/15-million-series-b), our team showed me that I was totally wrong. Here's why that was the case, what we did about it, and the results.

> >James Hawkins is PostHog Co-Founder & CEO. Connect with him on [LinkedIn](https://www.linkedin.com/in/j-hawkins/) and follow him on [Twitter](https://www.linkedin.com/in/j-hawkins/).

Author: No prob - could we focus on twitter only? more developer-y

> >James Hawkins is PostHog Co-Founder & CEO. Connect with him on [LinkedIn](https://www.linkedin.com/in/j-hawkins/) and follow him on [Twitter](https://www.linkedin.com/in/j-hawkins/).

Reviewer 3: ```suggestion
>James Hawkins is PostHog's Co-Founder & CEO. Connect with him on [LinkedIn](https://www.linkedin.com/in/j-hawkins/) and follow him on [Twitter](https://www.linkedin.com/in/j-hawkins/).
```

Idk if the "s" should be there or not, but sounds better like this to me

> ## Why I used to think this way

Reviewer 3: This section feels a bit jarring.

You have the most context, so I expect the compression ratio to be high: You understand exactly what you're trying to say here, simply based on a few key words.

It's not the same for me. I get where you're getting to, but I'd much rather prefer a short example showing this in action.

Here's an inferred expansion which I assume you mean (but can't confirm because I'm writing it, not you :P  - excuse my formatting/grammar/writing, just a quick example):

"I'd previously worked on building enterprise products with an enterprise sales team. This usually meant we always had this list of features that sales needed to close new customers, prioritised by their relative usefulness. Our goal was to build as many as we could.

There were invisible-to-me tradeoffs being made here: We'd get lots of initial enterprise contracts, but all these features didn't fit together coherently, which meant all these users would find the product frustrating to use, eventually leading to churn. Needless to say, there was no word-of-mouth growth"

.. and I realise now I didn't write it via an example but just as a story using the same bullets. lol.

Reviewer 2: I quite like the bullets.  They're good pacing and readability and easy to scan.

Reviewer 3: Fair, just mentioning my preference, but I understand it can be a minority opinion / not optimised for our goals :)

Reviewer 2: Reading it back now, I kind of see your point... it just needs some connective tissue so it flows a bit better.

```suggestion
## Why I used to think this way

My previous experience of product management was something like this:

* Here's a list of features the enterprise sales team wants
* How much pipeline does each of these features influence?
* _Builds as much of the above as possible whilst trying to manage existing customer requests_

This lead to lots of initial enterprise contracts, but negative outcomes in the long term like:

* A product users find frustrating to use
* No word of mouth growth
* Customer churn in the long term

These were all traps I wanted to avoid for PostHog, and lead me to favour hiring more engineering talent over product people.
```

Reviewer 2: Reading it back, I see your point. I've suggested a couple of small edits just to help it flow a little better.

> 
> My previous experience of product management:
> 
> * Here's a list of features the enterprise sales team wants
> * How much pipeline does each of these features influence?
> * _Builds as much of the above as possible whilst trying to manage existing customer requests_
> 
> Led to:
> 
> * Lots of initial enterprise contracts
> * A product users find frustrating to use
> * No word of mouth growth
> * Customer churn in the long term
> 

Reviewer 2: A few tweaks to Neil's point to help make this flow a little better.

```suggestion
## Why I used to think this way

My previous experience of product management was something like this:

* Here's a list of features the enterprise sales team wants
* How much pipeline does each of these features influence?
* _Builds as much of the above as possible whilst trying to manage existing customer requests_

This lead to lots of initial enterprise contracts, but negative outcomes in the long term like:

* A product users find frustrating to use
* No word of mouth growth
* High customer churn

These were all traps I wanted to avoid for PostHog and lead me to favour hiring more engineering talent over product people.

```

Reviewer 3: I think this is much better :)

> ## It starts with expectation setting

Reviewer 3: What is this section trying to say? I'm confused. There's a few gifs, and a mention of Goodharting, so I guess you're trying to say something like.... what you mention in the above section has even more negative consequences?

<a href="https://imgflip.com/i/61pr1n"><img src="https://i.imgflip.com/61pr1n.jpg" title="made at imgflip.com"/></a><div><a href="https://imgflip.com/memegenerator">from Imgflip Meme Generator</a></div><br />

Alas.

Whilst we should measure what matters – often it's what is measured, is what ends up mattering.

Revenue is pretty easy to measure and feels important. A B2B SaaS company will therefore default into prioritizing this. That feels intuitive - your company will die without money.

And you should sell more to get more revenue, right?

> <div style="width:360px;max-width:100%;"><div style="height:0;padding-bottom:65%;position:relative;"><iframe width="360" height="234" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameBorder="0" src="https://imgflip.com/embed/61rx1x"></iframe></div><p><a href="https://imgflip.com/gif/61rx1x">via Imgflip</a></p></div>

Reviewer 2: I'm inclined to remove the second meme. Two in the space of a paragraph or two feels a bit much.

Author: agreed

If product focuses on enabling existing pipeline to close, versus customer retention and growth, you end up in the above failure mode.

> ## How we approached building a small product team

Reviewer 3: This section is beautiful - love the points!

We've two product managers in a team of 30 people. [Results later](#the-difference-it-has-made) (they're good), here's what we did.

### 1. We hired product managers who can code

Our product people must be able to empathize with our users. We build for developers, so that meant hiring people that could write at least a little code. It felt like a compromise to not look for this.

> We looked for scrappy, anti-plan type folk who wouldn't create a detailed year long roadmap but would create loose guidance, with tons of context, and would help the company iterate its product over time. This fitted the culture we felt was optimal. Product managers who can code are likely to be a bit scrappier - after all, they've probably acquired both these skills by working in smaller organizations where roles are broad.

Author: ```suggestion
We looked for scrappy, anti-plan type folk who wouldn't create a detailed year long roadmap but would create loose guidance, with plenty of context, and would help the company iterate its product over time. This fitted the culture we felt was optimal. Product managers who can code are likely to be a bit scrappier - after all, they've probably acquired both these skills by working in smaller organizations where roles are broad.
```

Above all, though, the most important part of getting the team to click was our engineers telling Tim (my cofounder) and me that we needed to hire a product person. Sidenote: this happened in a [full team feedback session](../handbook/people/feedback#full-team-feedback-sessions). They were absolutely right.

### 2. We got our board to look at leading indicator metrics

We want to focus on optimizing leading indicator metrics, as high up the funnel as possible. That means you have a faster feedback loop for learning. We put those into our board meetings.

> PostHog still sends financial and revenue data each time, and has modelling to control the business' burn rate appropriately, but it goes into an appendix. We need to make sure this output connects to the input of our leading indiciators. If it starts diverging, look for different leading indicators.

Author: ```suggestion
PostHog still sends financial and revenue data each time, and has modelling to control the business' burn rate appropriately, but it goes into an appendix. We need to make sure this output connects to the input of our leading indicators. If it starts diverging, look for different leading indicators.
```

PostHog uses [small teams](../handbook/people/team-structure/why-small-teams), inspired by Twilio, who were inspired by Amazon. [Jeff Lawson's book](https://www.askyourdeveloper.com/) is a good 101 on this topic. TL;DR - we're 30 people, split into teams of no more than 6. The most crucial concept is that a team can ship into production without any dependencies. This optimizes for autonomy, not control.

These teams are supported by product people. That means engineers are not told in detail what to do, or what is ok to merge. We put our product people into specific small teams where there is lots of help needed, but many teams run fine without them doing more than informal support (at our current scale at least).

### 4. We created two (very) basic frameworks

We created a Nail X framework. You tell the company "Nail X" where X is a thing you must nail. I'm not joking. [Unnuanced communication can be powerful](https://twitter.com/danluu/status/1487228574608211969). Product people decide what we need to nail.

"Nail Funnels" was the first thing we did when Marcus (VP Product) joined last year. And we did. We made [Funnels](/product/funnels) in PostHog awesome.

> Get X reference customers was another useful framework. This came out of a board meeting. We realized that we had product market fit with our free open-source product, but we'd not yet worked on it for our PostHog Scale (paid product). We defined these as "genuinely delighted customers, using the product a lot, paying full price". We listed a few customers out we felt could achieve these three things to become a reference. Then we told the company to drop everything if needed to get it right with this group. Our product team made sure everyone was on track to achieve this.

Author: i'll add a link here as i blogged about this board meeting

### 5. We wrote stuff down

This matters to us as it prevents "information silos". I tried hard to find a less corporate way to describe that succinctly, but couldn't.

_Everyone_ must work like this. This is what the exec team at PostHog does:

* We share our exec meetings publicly in the company. Anyone can comment / read.
* We write out strategy changes into a pull request. Anyone can comment / read / suggest changes.
* We share our board meetings publicly in the company. Anyonen can comment / read.
* We share realtime updates around fundraising. Not after the fact.
* We have a public handbook - which is world editable

### 6. Product should involve engineers

Product are proactive with getting feedback. They involve engineering - many times a day, PostHog's engineering team speak to real life users. That may be over Slack, or it may be on a call.

> When engineers release new features, we'll encourage them to [collaborate directly with users](https://neilkakkar.com/How-I-Own-Projects-as-a-Software-Engineer.html) in real life.

Reviewer 4: I think it might be worth highlighting Neil's main point a bit more in this sentence.. e.g. something along the lines of....Engineers own solving problems end-to-end, product managers find and shape the biggest problems for the engineers to solve..

Author: agreed - added more context here and clarified it
